core/chain_makers.go

```diff
@@ -86,10 +85,10 @@ func (b *BlockGen) SetExtra(data []byte) {
 // added. Notably, contract code relying on the BLOCKHASH instruction
 // will panic during execution.
 func (b *BlockGen) AddTx(tx *types.Transaction) {
-	if b.coinbase == nil {
+	if b.gasPool == nil {
```

Comment:
Here too, checking the `gasPool` for `nil` and then setting the coinbase is very non-intuitive imho.